faas-samples/BearerBot/README.md

```diff
@@ -0,0 +1,220 @@
+
+---
+
+# Function Preparation
+
+This solution utilises a 'Bot' Type Conversational Cloud user, which logins to the platform with an API key, wo we'll need to make an API Key first, and then the Bot User.
+
+## Creating the Conversational Cloud API Key
+
+1. Within CC, navigate to _API keys_ `(campaigns -> data -> API keys)`
+2. Create a new API Key, name it 'FAAS-RUNNER'
+   The API key should have the following permissions enabled:
+   - login
+   - Conversation History / Messaging Interactions
+3. Make a note of the API key credentials (App key, Secret, Access Token, Access Token Secret) as we will need these in subequent steps
+
+## Creating the Conversational Cloud user
+
+1. Navigate to '_USERS_'
+2. Create a new Conversational Cloud user account with the following values:
+   - **login name**: 'FAAS-RUNNER' (use the same for fullName, nickName)
+   - **Roles**: Administrator and Agent Manager
+   - **Manager of**: "Main Group" (or top level agent group)
+   - **email address**: anything...
+   - **Login type**: API: use the `FAAS-RUNNER` API key we made in the previous step
+
+## Configuring the FAAS Settings
+
+1. From the _Quick Launch_ menu, open Functions
+2. In Functions, navigate to '_settings_'
+3. Open the '_secret storage_' tab, and click 'add a secret'
+4. Name the secret 'FAAS-RUNNER'
+5. From the below dropdown, select 'JSON'
+6. Now, using the API Key credentials (which we made a note of prior)
+
+```json
+{
+  "username": "FAAS-RUNNER",
+  "appKey": "{{app key}}",
+  "secret": "{{secret}}",
+  "accessToken": "{{access token}}",
+  "accessTokenSecret": "{{access token secret}}",
+  "bearer": null
+}
+```
+
+8. Save and exit settings
+
+## Create the function
+<<<<<<< HEAD
+=======
+1. Complete the *Coding Details* section with the following:
+   * **Event**: No Event
+   * **Template**: Greeting Template
+   * **Access to external domains?**: No
+>>>>>>> e886606605f1e5b413de33d064e3518c38cc847a
+
+1. Complete the _Coding Details_ section with the following:
+
+   - **Event**: No Event
+   - **Template**: Greeting Template
+   - **Access to external domains?**: Yes: On the following screen, add `*.context.liveperson.net` to ensure the context service URL is accessible from the function. Alternatively, the domain can be whitelisted from the FaaS settings page.
+
+   Click _Continue_
+
+2. Complete the _Function Description_ section with the following:
+
+<<<<<<< HEAD
+   - **Function Name**: currentQueueHealth
+   - **Description**: Retrieve current queue info on scheduled intervals and update a namespace in Conversation Context Service.
+=======
+2. Complete the *Function Description* section with the following:
+   * **Function Name**: faas-application-login
+   * **Description**: Performs application login for authentication token
+>>>>>>> e886606605f1e5b413de33d064e3518c38cc847a
```

Comment:
There appears to have been a merge conflict here that got pushed up. I assume lines 61-67 can just be deleted.

Reply:
Thanks @dahby - weird, appeared to have occured on a few other lines also. Just went through and cleaned those up.